### PR TITLE
useFilteredColumns argument was added

### PR DESCRIPTION
**Pull request by olhabn**

### CODE REVIEW

File: src/foam/u2/ExportModal.js [hunk 1]
```diff
@@ -99,6 +105,13 @@ foam.CLASS({
           .start(this.DATA_TYPE).end()
           .start().addClass('label').add('Response').end()
           .start(this.NOTE).addClass('input-box').addClass('note').end()
+          .add(
+            self.slot(function(dataType) {
+              if(dataType == 'CSV') {
```
Comment (kgrgreer) [inline src/foam/u2/ExportModal.js]: Follow FOAM style guide for spacing:
 if ( dataType == 'CSV' ) {

File: src/foam/nanos/export/CSVDriver.js [hunk 2]
```diff
@@ -22,10 +22,10 @@ foam.CLASS({
       outputter.outputFObject(X, obj);
       return outputter.toString();
     },
-    function exportDAO(X, dao) {
+    function exportDAO(X, dao, useFilteredColumns) {
```
Comment (kgrgreer) [inline src/foam/nanos/export/CSVDriver.js]: Do we need to add this? Couldn't we just update filteredTableColumns in the context?

Comment (olhabn) [reply src/foam/nanos/export/CSVDriver.js]: I thought that if we do that, we'll lose info about columns that was filtered by user, 
but I'll check my assumption

File: src/foam/u2/ExportModal.js (hunk 1 above)
Comment (olhabn) [reply src/foam/u2/ExportModal.js]: fixed

File: src/foam/nanos/export/CSVDriver.js (hunk 2 above)
Comment (olhabn) [reply src/foam/nanos/export/CSVDriver.js]: fixed

File: src/foam/u2/ExportModal.js [hunk 3]
```diff
@@ -99,6 +107,13 @@ foam.CLASS({
           .start(this.DATA_TYPE).end()
           .start().addClass('label').add('Response').end()
           .start(this.NOTE).addClass('input-box').addClass('note').end()
+          .add(
+            self.slot(function(dataType) {
+              if( dataType == 'CSV' ) {
```
Comment (kgrgreer) [inline src/foam/u2/ExportModal.js]: space after 'if'

File: src/foam/u2/ExportModal.js [hunk 4]
```diff
@@ -115,12 +130,19 @@ foam.CLASS({
         return;
       }
 
+      var filteredColumnsCopy = this.filteredTableColumns;
+      if( ! this.useFiltered )
```
Comment (kgrgreer) [inline src/foam/u2/ExportModal.js]: space after 'if'

File: src/foam/u2/ExportModal.js [hunk 5]
```diff
@@ -115,12 +130,19 @@ foam.CLASS({
...
+        this.filteredTableColumns$.set(null);
+
       var exportDriver = await this.exportDriverRegistryDAO.find(this.dataType);
       exportDriver = foam.lookup(exportDriver.driverName).create();
 
       this.note = this.exportData ?
         await exportDriver.exportDAO(this.__context__, this.exportData) :
         await exportDriver.exportFObject(this.__context__, this.exportObj);
+
+        if( ! this.useFiltered )
```
Comment (kgrgreer) [inline src/foam/u2/ExportModal.js]: space after 'if'

File: src/foam/u2/ExportModal.js (hunk 3 above)
Comment (olhabn) [reply src/foam/u2/ExportModal.js]: Changes made.
Please review

File: src/foam/u2/ExportModal.js (hunk 4 above)
Comment (olhabn) [reply src/foam/u2/ExportModal.js]: fixed

File: src/foam/u2/ExportModal.js (hunk 5 above)
Comment (olhabn) [reply src/foam/u2/ExportModal.js]: fixed

**Review by jlhughes — APPROVED** [review]
All comments resolved.

File: src/foam/u2/ExportModal.js [hunk 6]
```diff
@@ -115,12 +129,19 @@ foam.CLASS({
         return;
       }
 
+      var filteredColumnsCopy = this.filteredTableColumns;
+      if ( this.exportAllColumns )
+        this.filteredTableColumns$.set(null);
```
Comment (kgrgreer) [inline src/foam/u2/ExportModal.js]: Couldn't you just do this.filteredTableColumns = null; ?

File: src/foam/u2/ExportModal.js [hunk 7]
```diff
@@ -144,6 +169,9 @@ foam.CLASS({
         document.body.appendChild(link);
         link.click();
       })
+
+      if ( this.exportAllColumns )
+        this.filteredTableColumns$.set(filteredColumnsCopy);
```
Comment (kgrgreer) [inline src/foam/u2/ExportModal.js]: Couldn't you just do this.filteredTableColumns = null; ?

File: src/foam/u2/ExportModal.js (hunk 6 above)
Comment (olhabn) [reply src/foam/u2/ExportModal.js]: My mistake

File: src/foam/u2/ExportModal.js (hunk 7 above)
Comment (olhabn) [reply src/foam/u2/ExportModal.js]: fixed

**Pull request merged**